cmd/skopeo/sync.go

```diff
@@ -277,16 +359,26 @@ func syncHandler(c *cli.Context) (retErr error) {
 	//TODO: should we assume that's our default manifest type?
 	manifestType := manifest.DockerV2Schema2MediaType
 
-	toCopy := imagesBySource{}
+	if c.IsSet("src-cert-dir") || c.IsSet("src-tls-verify") || c.IsSet("src-creds") {
+		return fmt.Errorf("Must either specify a '--source-file' or '--src-' flags")
+	}
```

Comment:
These lines have to be moved into `func syncSourceFileHandler`  otherwise they will break the cli when `--source-file` is not set.

Reply:
oops...!